GenderPayGap.WebUI/Models/Admin/AdminUserResendVerificationEmailController.cs

```diff
@@ -72,21 +74,26 @@ public IActionResult ResendVerificationEmailPost(long id, AdminResendVerificatio
                     viewModel.Reason
                 },
                 currentUser);
+            
+            string verificationCode = Guid.NewGuid().ToString("N");
+            string verificationUrl = Url.Action(
+                "VerifyEmail",
+                "AccountCreation",
+                new {code = verificationCode},
+                "https");
+
+            try
+            {
+                EmailSendingService.SendAccountVerificationEmail(user.EmailAddress, verificationUrl);
+                user.EmailVerifyHash = verificationCode;
+                user.EmailVerifySendDate = VirtualDateTime.Now;
 
-            string verifyCode = Encryption.EncryptQuerystring(user.UserId + ":" + user.Created.ToSmallDateTime());
-
-            user.EmailVerifyHash = Crypto.GetSHA512Checksum(verifyCode);
-            user.EmailVerifySendDate = VirtualDateTime.Now;
-            dataRepository.SaveChangesAsync().Wait();
-
-            string verifyUrl = Url.Action("VerifyEmail", "Register", new { code = verifyCode }, "https");
-
-            if (!Emails.SendCreateAccountPendingVerificationAsync(verifyUrl, user.EmailAddress).Result)
+                dataRepository.SaveChangesAsync().Wait();
+            }
+            catch
             {
-                viewModel.AddErrorFor(
-                    m => m.OtherErrorMessagePlaceholder,
-                    "Error whilst re-sending verification email. Please try again in a few minutes.");
-                return View("ResendVerificationEmail", viewModel);
+                // help user resend email
+                throw new Exception("Failed to send verification email. Please try again");

```

Comment:
I don't think the user will see the message of this error.
I think if we throw an exception, they are shown the generic "Something went wrong" message.
In which case, it might be better to **not** catch the exception, so we can see what the original exception is (e.g. some database exception)

Reply:
Done. I've now removed the try/catch block